.github/workflows/main.yml

```diff
@@ -31,9 +30,39 @@ jobs:
         env:
           CI: false
         run: yarn build
-      - name: Run test
+      - name: Run tests
         run: yarn test
-      - name: Execute format
-        run: yarn format
+
+  lint:
+    name: Lint code
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v3
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+      - name: Check Yarn version
+        run: yarn --version
+      - name: Install dependencies
+        run: yarn install
       - name: Execute lint
         run: yarn lint
+
+  format:
+    name: Format code
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v3
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+      - name: Check Yarn version
+        run: yarn --version
+      - name: Install dependencies
+        run: yarn install
```

Comment:
```suggestion
```

Reply:
Let's run lint and format in the same job. No need to install deps twice. Test should run in parallel though.

Reply:
okay